src/realm/alloc_slab.hpp

```diff
@@ -300,6 +300,36 @@ class SlabAlloc : public Allocator {
     void do_free(ref_type, const char*) noexcept override;
     char* do_translate(ref_type) const noexcept override;
 
+    /// Returns the first section boundary *above* the given position.
+    size_t get_upper_section_boundary(size_t start_pos) const noexcept;
```

Comment:
@finnschiermer what do you think about changing these `SlabAlloc` functions to take and return `uint64_t`?